src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs

```diff
@@ -491,7 +495,13 @@ impl CloudHypervisorInner {
         } else if confidential_guest {
             self.guest_protection_to_use = protection.clone();
 
-            info!(sl!(), "guest protection available and requested"; "guest-protection" => protection.to_string());
+            if protection == GuestProtection::Tdx {
+                info!(sl!(), "guest protection available and requested"; "guest-protection" => protection.to_string());
+            } else {
+                return Err(anyhow!(GuestProtectionError::ExpectedTDXProtection(
+                    protection
+                )));
+            }
         } else if protection == GuestProtection::Tdx {
```

Comment:
We are checking for `GuestProtection::Tdx` here and on line 498 as well. Similarly the check for `confidential_guest` is made on line 490 and then 495. Its a bit confusing on how the checks are made. I think we can organize this better with checking for  confidential_guest first and then branching out from there:

```
if  confidential_guest {
         if protection == GuestProtection::NoProtection {
         } 
         else if protection == GuestProtection::Tdx {
          } 
         else { 
          } 
 } else {
     // We could omit this branch altogether as protection value does not really matter if confidential case if not enforced
}
```

Reply:
@amshinde - Agreed. Branch updated.